pkg/controller/kubelet-config/kubelet_config_controller.go

```diff
@@ -473,6 +474,18 @@ func (ctrl *Controller) syncKubeletConfig(key string) error {
 		klog.V(4).Infof("Finished syncing kubeletconfig %q (%v)", key, time.Since(startTime))
 	}()
 
+	if err := wait.PollUntilContextTimeout(context.TODO(), ctrlcommon.ControllerConfigRolloutInterval, ctrlcommon.ControllerConfigTimeout, false, func(_ context.Context) (bool, error) {
+		_, err := ctrl.client.MachineconfigurationV1().ControllerConfigs().Get(context.TODO(), ctrlcommon.ControllerConfigName, metav1.GetOptions{})
```

Comment:
Do you think reading off the ccLister here instead would work instead of a direct API fetch?

If not, maybe we can tone down the `ControllerConfigRolloutInterval` a bit to reduce the API requests

Reply:
Also curious, does this help with upgrades? The object should always exist after install time. Or I guess this is meant to fix installs only

Reply:
It may be worth a CI cycle just to see if we can get away with using the lister instead. Although I don't know how difficult this particular issue is to reproduce there.